io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -75,6 +82,30 @@ impl Display for SubType {
 pub struct NvmfSubsystem(pub(crate) NonNull<spdk_nvmf_subsystem>);
 pub struct NvmfSubsystemIterator(*mut spdk_nvmf_subsystem);
 
+#[repr(C)]
+pub struct SpdkNvmfController(pub(crate) NonNull<spdk_nvmf_ctrlr>);
+
+impl SpdkNvmfController {
+    fn get_hostnqn(&self) -> String {
+        unsafe { self.0.as_ref().hostnqn.as_str().to_string() }
+    }
+}
+impl From<*mut spdk_nvmf_ctrlr> for SpdkNvmfController {
+    fn from(s: *mut spdk_nvmf_ctrlr) -> Self {
+        SpdkNvmfController(NonNull::new(s).unwrap())
+    }
+}
+
+impl Debug for SpdkNvmfController {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        unsafe {
+            f.debug_struct("SpdkNvmfController")
+                .field("hostnqn", &self.0.as_ref().hostnqn.as_str().to_string())
+                .finish()
+        }
```

Comment:
Reduce the unsafe code scope to what's necessary.
```suggestion
        let hostnqn = unsafe { self.get_hostnqn() };
            f.debug_struct("SpdkNvmfController")
                .field("hostnqn", &hostnqn)
                .finish()
```

Reply:
a good suggestion to use get_hostnqn, but it doesn't need the unsafe block.